src/main/hook/patch/gfx.c

```diff
@@ -191,17 +280,60 @@ XVisualInfo *glXChooseVisual(Display *dpy, int screen, int *attribList)
   return res;
 }
 
+// Check for fullscreen draw calls and scale them appropriately.
```

Comment:
So, we have two places that require scaling, one being the viewport and another one here. We discussed this somewhat, but seeing the solution now, I have my doubts this is the better solution compared to scaling the final framebuffer. 

There was something with how some games render videos. Was that the problem the frame buffer solution doesn't work? If yes, please add this reasoning as a comment here. I consider this an important technical design decision that deserves explanation in the code. So someone else/future me isn't puzzled by this again.